lib/Conversion/StableHLOToTTIR/StableHLOToTTIRPatterns.cpp

```diff
@@ -749,17 +749,63 @@ class StableHLOToTTIRBroadcastInDimOpConversionPattern
+
     auto outputType = mlir::cast<RankedTensorType>(
         getTypeConverter()->convertType(srcOp.getResult().getType()));
-    tensor::EmptyOp outputTensor = rewriter.create<tensor::EmptyOp>(
-        srcOp.getLoc(), outputType.getShape(), outputType.getElementType());
 
-    mlir::ArrayAttr dimArg =
-        rewriter.getI64ArrayAttr(adaptor.getBroadcastDimensions());
+    if (inputType.getRank() == outputType.getRank()) {
+      // unsqueeze is not needed, proceed to converting to broadcast
+      tensor::EmptyOp outputTensor = rewriter.create<tensor::EmptyOp>(
+          srcOp.getLoc(), outputType.getShape(), outputType.getElementType());
+
+      mlir::ArrayAttr dimArg =
+          rewriter.getI64ArrayAttr(adaptor.getBroadcastDimensions());
 
-    rewriter.replaceOpWithNewOp<mlir::tt::ttir::BroadcastOp>(
-        srcOp, getTypeConverter()->convertType(outputTensor.getType()),
-        Value(adaptor.getOperand()), Value(outputTensor), dimArg);
+      rewriter.replaceOpWithNewOp<mlir::tt::ttir::BroadcastOp>(
+          srcOp, getTypeConverter()->convertType(outputTensor.getType()),
+          adaptor.getOperand(), outputTensor, dimArg);
+    } else {
+      // This stablehlo operation cannot be represented in a single TTIR.
+      // It has to be split into ttir.reshape followed by a ttir.broadcast op.
+      std::vector<int64_t> unsqueezeShape(outputType.getRank());
+      ::llvm::ArrayRef<int64_t> broadcast_in_dim =
+          adaptor.getBroadcastDimensions();
+      std::fill(unsqueezeShape.begin(), unsqueezeShape.end(), 1);
+
+      for (int64_t i = 0; i < inputType.getRank(); i++) {
+        unsqueezeShape[broadcast_in_dim[i]] = inputType.getDimSize(i);
+      }
+
+      RankedTensorType unsqueezeOutputType =
+          RankedTensorType::get(unsqueezeShape, outputType.getElementType());
+
+      tensor::EmptyOp reshapeOutputTensor = rewriter.create<tensor::EmptyOp>(
+          srcOp.getLoc(), unsqueezeOutputType.getShape(),
+          unsqueezeOutputType.getElementType());
+
+      SmallVector<int32_t> reshapeDim(unsqueezeShape.begin(),
+                                      unsqueezeShape.end());
+      auto reshapeDimAttr = rewriter.getI32ArrayAttr(reshapeDim);
+
+      mlir::tt::ttir::ReshapeOp reshape =
+          rewriter.create<mlir::tt::ttir::ReshapeOp>(
+              srcOp.getLoc(),
+              getTypeConverter()->convertType(unsqueezeOutputType),
+              adaptor.getOperand(), reshapeOutputTensor, reshapeDimAttr);
+
+      tensor::EmptyOp broadcastOutputTensor = rewriter.create<tensor::EmptyOp>(
+          srcOp.getLoc(), outputType.getShape(), outputType.getElementType());
+
+      mlir::ArrayAttr dimArg =
+          rewriter.getI64ArrayAttr(adaptor.getBroadcastDimensions());
+
+      rewriter.replaceOpWithNewOp<mlir::tt::ttir::BroadcastOp>(
+          srcOp,
+          getTypeConverter()->convertType(broadcastOutputTensor.getType()),
+          reshape.getResult(), broadcastOutputTensor, dimArg);
```

Comment:
It is not used yet, so I considered removing it, but then decided to keep this for future in case this information is required at TTIR level for any optimization.